src/stub_launcher.cc

```diff
@@ -0,0 +1,584 @@
+        int status;
+        waitpid((*stub_pid), &status, 0);
+        (*stub_pid) = 0;
+        throw BackendModelException(
+            TRITONSERVER_ErrorNew(TRITONSERVER_ERROR_INTERNAL, ex.what()));
+      }
+    } else if (stub_process_kind_ == "INSTANCE_STUB") {
+      RETURN_IF_ERROR(InstanceStubProcess(
+          stub_message_queue, parent_message_queue, shm_pool));
+    } else {
+      return TRITONSERVER_ErrorNew(
+          TRITONSERVER_ERROR_INTERNAL,
+          (std::string("Unknown stub_process_kind: ") + stub_process_kind_)
+              .c_str());
+    }
+
+    initialized_ = true;
+  }
+
+  return nullptr;
+}
+
+void
+StubLauncher::ModelStubProcess(
+    std::unique_ptr<MessageQueue<bi::managed_external_buffer::handle_t>>*
+        stub_message_queue,
+    std::unique_ptr<MessageQueue<bi::managed_external_buffer::handle_t>>*
+        parent_message_queue,
+    std::unique_ptr<SharedMemoryManager>* shm_pool)
+{
+  triton::common::TritonJson::WriteBuffer buffer;
+  model_state_->ModelConfig().Write(&buffer);
+
+  std::string model_config = buffer.MutableContents();
+
+  std::unique_ptr<IPCMessage> auto_complete_message =
+      IPCMessage::Create((*shm_pool), false /* inline_response */);
+  auto_complete_message->Command() = PYTHONSTUB_AutoCompleteRequest;
+
+  std::unique_ptr<PbString> pb_string =
+      PbString::Create((*shm_pool), model_config);
+  bi::managed_external_buffer::handle_t string_handle = pb_string->ShmHandle();
+
+  auto_complete_message->Args() = string_handle;
+  (*stub_message_queue)->Push(auto_complete_message->ShmHandle());
+
+  std::unique_ptr<IPCMessage> auto_complete_response_message =
+      IPCMessage::LoadFromSharedMemory(
+          (*shm_pool), (*parent_message_queue)->Pop());
+
+  if (auto_complete_response_message->Command() !=
+      PYTHONSTUB_AutoCompleteResponse) {
+    throw PythonBackendException(
+        "Received unexpected resposne from Python backend stub: " + name_);
+  }
+
+  auto auto_complete_response =
+      std::move(((*shm_pool)->Load<AutoCompleteResponseShm>(
+                    auto_complete_response_message->Args())))
+          .data_;
+
+  if (auto_complete_response->response_has_error) {
+    if (auto_complete_response->response_is_error_set) {
+      std::unique_ptr<PbString> error_message = PbString::LoadFromSharedMemory(
+          (*shm_pool), auto_complete_response->response_error);
+      throw PythonBackendException(error_message->String());
+    } else {
+      throw PythonBackendException("Auto-complete failed for " + name_);
+    }
+  }
+
+  if (auto_complete_response->response_has_model_config) {
+    std::unique_ptr<PbString> auto_complete_config =
+        PbString::LoadFromSharedMemory(
+            (*shm_pool), auto_complete_response->response_model_config);
+    std::string auto_complete_config_string = auto_complete_config->String();
+    if (!auto_complete_config_string.empty()) {
+      FixStringToJsonFormat(&auto_complete_config_string);
```

Comment:
Because in the JSON string returned from model.py, python is using "True/False", but in c++ side, we need to use "true/false" to make it parsed successfully.

Reply:
I left a new comment above on why I think this shouldn't be required. I think the correct json format doesn't accept True/False as valid boolean values.